python-package/xgboost/dask/__init__.py

```diff
@@ -506,13 +506,46 @@ async def map_worker_partitions(
                 args.append(ref._create_fn_args(addr))
             else:
                 args.append(ref)
+
+        @wraps(func)
+        def fn(
+            *args: _P.args, address: str = addr, **kwargs: _P.kwargs
+        ) -> List[_MapRetT]:
+            # Turn result into a list for bag construction
+            worker = distributed.get_worker()
+
+            if worker.address != address:
+                raise ValueError(
+                    f"Invalid worker address: {worker.address}, expecting {address}. "
+                    "This is likely caused by one of the workers died and Dask "
+                    "re-scheduled a different one. Resilience is not yet supported."
+                )
+            return [func(*args, **kwargs)]
```

Comment:
Will the functions that run on existing workers wait indefinitely for the dead worker to appear, or will they start failing at some point?